sw/device/tests/aon_timer_irq_test.c

```diff
@@ -211,11 +214,15 @@ bool test_main(void) {
   // frequency to make sure the aon timer is generating the interrupt after the
   // choosen time and there's no error in the reference time measurement. This
   // calculation is required as the various platforms used for testing have
-  // differing clocks frequencies. A few hundred cycles is required for the
-  // interrupt to note the elapsed time so the test fails with an unacceptable
-  // time variance when the sleep time is too low.
-  uint64_t low_time_range = ((1000000 * 500) / kClockFreqCpuHz) * 20;
-  uint64_t high_time_range = ((1000000 * 500) / kClockFreqCpuHz) * 40;
+  // differing clocks frequencies. A minimum amount of cycles is required for
+  // the interrupt to note the elapsed time so the test fails with an
+  // unacceptable time variance when the sleep time is too low.
+  enum{
+    kMinCycles = 30 * 1000,
+    kMaxCycles = 45 * 1000,
```

Comment:
Sorry to repeat myself, but can you explain where the magic "30" and "45" numbers come from?

Reply:
Sorry for not understanding your question. 
They are basically guessing, if we pick a bigger number the test will take too long, if we pick smaller numbers the overhead becomes a significant portion of the test time, so I did several iterations until I got those numbers.
Maybe the correct way is to try to measure the overhead to come with exact numbers.

Reply:
Hmm, ok. Seems a bit prone to things breaking: naively, I'd hope that this is something we could derive from first principles. (But maybe that's unrealistic, I'm not sure).